src/Raven.Server/Documents/Indexes/Persistence/Lucene/Documents/JintLuceneDocumentConverter.cs

```diff
@@ -69,11 +72,59 @@ protected override int GetFields<T>(T instance, LazyStringValue key, object docu
                 object value;
                 if (obj.IsObject() && obj.IsArray() == false)
                 {
+                    //Note that for dynamic fields 'field.Name' will contain the real name of the field and not 'property'
                     var val = TryDetectDynamicFieldCreation(property, obj.AsObject(), field);
                     if (val != null)
                     {
-                        value = GetValue(val);
-                        newFields += GetRegularFields(instance, field, value, indexContext);
+                        if (val.IsObject() && val.AsObject().TryGetValue("$spatial", out var dynamicSpatial))
+                        {
+                            obj = val;
+                        }
+                        else
+                        {
+                            value = GetValue(val);
+                            newFields += GetRegularFields(instance, field, value, indexContext);
+                            continue;
+                        }
+                    }
+
+                    if (obj.AsObject().TryGetValue("$spatial", out var inner))
+                    {
+                        if (inner.IsObject() == false)
+                        {
+                            throw new InvalidSpatialArgument($"Property {field.Name} is a spatial field but its value is not a valid spatial value.");
+                        }
+
+                        var sf = inner.AsObject();
+                        if (sf.TryGetValue("Type", out var type) == false || type.IsString() == false)
+                        {
+                            throw new InvalidSpatialArgument($"Property {field.Name} is a spatial field but it doesn't have a valid type.");
+                        }
+
+                        SpatialField spatialField = null;
+                        IEnumerable<AbstractField> spatial = null;
+                        switch (type.AsString())
+                        {
+                            case "Coordinates":
+                                if (sf.TryGetValue("Coordinates", out var coordinates) == false || coordinates.IsObject() == false)
+                                    throw new InvalidSpatialArgument($"Property {field.Name} is a spatial field of Coordinates type but has invalid or missing Coordinates field.");
+                                var coordinatesObj = coordinates.AsObject();
+                                if (coordinatesObj.TryGetValue("Lat", out var lat) == false || coordinatesObj.TryGetValue("Lng", out var lng) == false)
```

Comment:
In C#, we just ignore this scenario, I think this might be a better option here.

Reply:
what do you mean by "this scenario"? bad spatial field needs to be ignored and not error?

Reply:
Yes